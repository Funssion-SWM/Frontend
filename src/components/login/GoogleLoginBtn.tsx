'use client';

import Link from 'next/link';
import googleIcon from '@/assets/icons/google_icon.svg';

import Image from 'next/image';

export default function GoogleLoginBtn() {
  return (
    <Link
      href={`${process.env.NEXT_PUBLIC_SERVER_IP_ADDRESS_SECURE}/oauth2/authorization/google`}
      className="w-full"
    >
      <button
        className={`flex w-full justify-between items-center px-3.5 py-2 rounded-3xl transition border-2 border-soma-grey-30 hover:bg-soma-grey-20 text-sm sm:text-base`}
        onClick={() => {}}
      >
        <Image src={googleIcon} alt="googleIcon" width={20} height={20} />
        <p>구글 계정 로그인</p>
        <div className="w-5 h-5"></div>
      </button>
    </Link>
  );
}
