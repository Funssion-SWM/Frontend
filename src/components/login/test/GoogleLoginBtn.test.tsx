import renderer from 'react-test-renderer';
import GoogleLoginBtn from '../GoogleLoginBtn';

describe('GoogleLoginBtn', () => {
  it('renders correctly', () => {
    const component = renderer.create(<GoogleLoginBtn />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
