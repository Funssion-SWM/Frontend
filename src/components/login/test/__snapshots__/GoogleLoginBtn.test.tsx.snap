// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleLoginBtn renders correctly 1`] = `
<a
  className="flex w-full justify-between items-center px-3.5 py-2 rounded-3xl transition border-2 border-soma-grey-30 hover:bg-soma-grey-20 text-sm sm:text-base"
  href="undefined/oauth2/authorization/google"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <img
    alt="googleIcon"
    data-nimg="1"
    decoding="async"
    height={20}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
    srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={20}
  />
  <p>
    구글 계정 로그인
  </p>
  <div
    className="w-5 h-5"
  />
</a>
`;
